### PR TITLE
feat(alert): create alert component

### PR DESCRIPTION
**Pull request by SiTaggart**
- New status tokens to support the alert
- update media-object to use box so it can be used in alert
- create the alert component
- add alert to core package

### CODE REVIEW

File: packages/paste-core/components/alert/__tests__/index.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+      const wrapper: ReactWrapper = mount(<Alert variant="error">This is an alert</Alert>);
+      expect(wrapper.exists('[role="alert"]')).toEqual(true);
+    });
+
+    it('Should add the role of alert to the warning alert', () => {
+      const wrapper: ReactWrapper = mount(<Alert variant="warning">This is an alert</Alert>);
+      expect(wrapper.exists('[role="alert"]')).toEqual(true);
+    });
+
+    it('Should add the provided role of alert to the alert', () => {
```
Comment (TheSisb) [inline packages/paste-core/components/alert/__tests__/index.spec.tsx]: `'Should add the provided role of alert to the alert` -> `'Should add the provided role to the alert`?

File: packages/paste-core/components/alert/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+export type BackgroundColors = ValueOf<typeof BACKGROUND_COLORS>;
+export type BorderColors = ValueOf<typeof BORDER_COLORS>;
+export type Roles = ValueOf<typeof ROLES>;
+
+export interface AlertProps {
+  id?: never;
+  className?: never;
+  children: NonNullable<React.ReactNode>;
+  onDismiss?: () => void;
+  role?: string;
```
Comment (TheSisb) [inline packages/paste-core/components/alert/src/index.tsx]: Just curious, this can't be Roles because there are possibly other types of roles that would work here?

File: packages/paste-core/components/alert/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    return ROLES.error;
+  }
+  return ROLES.neutral;
+};
+
+const Alert: React.FC<AlertProps> = ({children, onDismiss, variant, role, ...props}) => {
+  return (
+    <Box
+      {...safelySpreadBoxProps(props)}
+      backgroundColor={getBackgroundColor(variant)}
```
Comment (TheSisb) [inline packages/paste-core/components/alert/src/index.tsx]: Since we share the same keys among all the objects defined above on lines 12-31, we can do something like this and remove all these functions/switches

```js
backgroundColor={BACKGROUND_COLORS[variant]}
borderColor={BORDER_COLORS[variant]}
role={role != null ? role : ROLES[variant]}
```
Typescript should make this safe to do, but additionally we can define the objects like this
```js
const BackgroundColors = {
 [ERROR]: 'colorBackgroundErrorLightest'
  ...
}
```
in JS land to make sure the keys matched.  We don't need that here though

File: packages/paste-core/components/alert/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+import {ValueOf} from '@twilio-paste/types';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {MediaObject, MediaFigure, MediaBody} from '@twilio-paste/media-object';
+import {Button} from '@twilio-paste/button';
+import {CloseIcon} from '@twilio-paste/icons/esm/CloseIcon';
+import {ErrorIcon} from '@twilio-paste/icons/esm/ErrorIcon';
+import {InformationIcon} from '@twilio-paste/icons/esm/InformationIcon';
+import {WarningIcon} from '@twilio-paste/icons/esm/WarningIcon';
+
+export const ROLES = {
```
Comment (TheSisb) [inline packages/paste-core/components/alert/src/index.tsx]: mega nit: these constants should be PascalCase and the Keys all UpperCased for consistency (paste-website/src/constants.ts)

File: packages/paste-core/core-bundle/src/index.tsx [hunk 5]
```diff
@@ -1,4 +1,5 @@
 export * from '@twilio-paste/absolute';
+export * from '@twilio-paste/alert';
```
Comment (TheSisb) [inline packages/paste-core/core-bundle/src/index.tsx]: I'm concerned `core` may not work this way anymore, since we're exporting `VARIANTS` from the Alert package, and undoubtedly another component will export Variants too.  Either we prefix all exports with the component name (i.e.: `AlertVariants`) in the component packages, or we manually maintain this file to do the renaming.  Thoughts?

File: packages/paste-core/utilities/media-object/src/index.tsx [hunk 6]
```diff
@@ -1,50 +1,45 @@
...
+  verticalAlign?: 'center' | 'top';
 }
 
-const MediaObject = styled.div<MediaObjectProps>`
-  align-items: ${(props: MediaObjectProps) => (props.isCentered ? 'center' : 'flex-start')};
-  display: flex;
-`;
+const MediaObject: React.FC<MediaObjectProps> = ({children, verticalAlign = 'top'}) => {
+  return (
+    <Box display="flex" alignItems={verticalAlign === 'top' ? 'flex-start' : 'center'}>
```
Comment (TheSisb) [inline packages/paste-core/utilities/media-object/src/index.tsx]: Should we use the Flex component here? 🤔

File: packages/paste-core/utilities/media-object/src/index.tsx [hunk 7]
```diff
@@ -1,50 +1,45 @@
...
-const getMargin = (props: MediaFigureProps): string => {
-  const direction = props.align === 'end' ? 'left' : 'right';
-  const spacing = themeGet(`space.${props.spacing}`)(props);
-  return `margin-${direction}: ${spacing}`;
+const MediaFigure: React.FC<MediaFigureProps> = ({children, align, spacing}) => {
+  return (
+    <Box
+      display="flex"
+      flexShrink={0}
+      marginLeft={align === 'end' ? spacing : undefined}
```
Comment (TheSisb) [inline packages/paste-core/utilities/media-object/src/index.tsx]: I don't think the Flex component currently allows margins, which may be an oversight after looking at this..

File: packages/paste-design-tokens/tokens/global/background-color.yml [hunk 8]
```diff
@@ -27,6 +27,9 @@ props:
     comment: Background color of the main page body
 
   # status
+  color-background-neutral-lightest:
+    value: "{!palette-blue-10}"
+    comment: Lightest neutral background color
```
Comment (TheSisb) [inline packages/paste-design-tokens/tokens/global/background-color.yml]: Should we be more explicit about these tokens, saying that they're used in Alerts and that they are paired with certain other colors?

File: packages/paste-design-tokens/tokens/global/text-color.yml [hunk 9]
```diff
@@ -55,16 +55,22 @@ props:
...
     value: "{!palette-orange-60}"
     comment: Color for warning text.
+  color-text-warning-dark:
+    value: "{!palette-orange-70}"
+    comment: Color for dark warning text.
 
   #icon
   color-text-icon:
-    value: "{!palette-gray-100}"
+    value: "{!palette-gray-70}"
```
Comment (TheSisb) [inline packages/paste-design-tokens/tokens/global/text-color.yml]: Just curious about this change, `color-text` still looks to be set to `gray-100`.  Why was this changed?

File: packages/paste-core/components/alert/src/index.tsx (hunk 2 above)
Comment (SiTaggart) [reply packages/paste-core/components/alert/src/index.tsx]: Yup, it's basically a free for all is you really want to do still things other than making alerts and status'

File: packages/paste-core/utilities/media-object/src/index.tsx (hunk 6 above)
Comment (SiTaggart) [reply packages/paste-core/utilities/media-object/src/index.tsx]: Thought about it, but there reason I didn't was that Alert would require Media-object as a peer, then that has a peer on Flex, and that has a peer on Box, and I use _so little_ of Flex it just didn't seem worth it

File: packages/paste-core/components/alert/src/index.tsx (hunk 3 above)
Comment (SiTaggart) [reply packages/paste-core/components/alert/src/index.tsx]: oooh yeah, that's much better

File: packages/paste-core/core-bundle/src/index.tsx (hunk 5 above)
Comment (SiTaggart) [reply packages/paste-core/core-bundle/src/index.tsx]: Switching all the exports, for sure this is going to be a problem

File: packages/paste-design-tokens/tokens/global/background-color.yml (hunk 8 above)
Comment (SiTaggart) [reply packages/paste-design-tokens/tokens/global/background-color.yml]: We should, and we have a ticket in the backlog to look at token meta data and pairing information. I'd punt to making these more desccriptive and detailed then so we can decide what we are expecting for all tokens

File: packages/paste-design-tokens/tokens/global/text-color.yml (hunk 9 above)
Comment (SiTaggart) [reply packages/paste-design-tokens/tokens/global/text-color.yml]: Good catch, total oversight.

File: packages/paste-design-tokens/tokens/themes/console/global/text-color.yml [hunk 10]
```diff
@@ -56,16 +56,22 @@ props:
...
     value: "{!palette-orange-60}"
     comment: Color for warning text.
+  color-text-warning-dark:
+    value: "{!palette-orange-70}"
+    comment: Color for dark warning text.
 
   #icon
   color-text-icon:
-    value: "{!palette-gray-100}"
+    value: "{!palette-gray-70}"
```
Comment (TheSisb) [inline packages/paste-design-tokens/tokens/themes/console/global/text-color.yml]: And this one too?

File: packages/paste-design-tokens/tokens/themes/sendgrid/global/text-color.yml [hunk 11]
```diff
@@ -53,16 +53,22 @@ props:
...
     value: "{!palette-orange-60}"
     comment: Color for warning text.
+  color-text-warning-dark:
+    value: "{!palette-orange-70}"
+    comment: Color for dark warning text.
 
   #icon
   color-text-icon:
-    value: "{!palette-gray-100}"
+    value: "{!palette-gray-70}"
```
Comment (TheSisb) [inline packages/paste-design-tokens/tokens/themes/sendgrid/global/text-color.yml]: 70 -> 100 too?

File: packages/paste-design-tokens/tokens/themes/console/global/text-color.yml (hunk 10 above)
Comment (SiTaggart) [reply packages/paste-design-tokens/tokens/themes/console/global/text-color.yml]: Oh, no these are to spec for both console and sendgrid. So I'm going to put it back for the global theme

**Comment by richbachman** [comment]
🚢
